src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -117,31 +118,26 @@ public User getUser(String email) {
   /**
   * Returns a set of all users who have posted a message or about me
   */
-  public Set<String> getUsers()
+  public HashMap<String, String> getUsers()
   {
-    Set<String> users = new HashSet<>();
-    Set<String> usersWithAboutMe = new HashSet<>();
-	
-    /*adding users who have added about me*/
-    Query userQuery = new Query("User");
-    PreparedQuery userQueryResults = datastore.prepare(userQuery);
-    for(Entity entity: userQueryResults.asIterable())
-    {
-      users.add((String)entity.getProperty("email") +" "+ (String)entity.getProperty("aboutMe"));
-      usersWithAboutMe.add((String)entity.getProperty("email"));
-    }
-	
-    /*adding users who have posted a message*/
-    Query messageQuery = new Query("Message");
-    PreparedQuery messageQueryResults = datastore.prepare(messageQuery);
-    for(Entity entity: messageQueryResults.asIterable())
-    {
-      if(!usersWithAboutMe.contains((String)entity.getProperty("user")))
+      HashMap<String, String> usersMap = new HashMap<>();
+
```

Comment:
I used a HashMap<String(user link), String(user-link+"|"+aboutMe)> instead of two HashSets as you suggested. The order of messageQuery and userQuery does not matter, but the code will be slightly different based on the order of the two queries. if messageQuery is first (as in the code) the user-link+"|" will be added without aboutMe text because Message does not have that property, but later with userQuery if the same user has posted aboutMe, then it would be a duplicate, which map does not allow for that user-link key, but the value for that user key (user-link+"|") will be replaced by user-link+"|"+aboutMe. If userQuery was first, then we would need to check if HashMap contains that user, to avoid over-writing of value user-link+"|"+aboutMe by user-link+"|" if the same user comes up in messageQuery.